native/common/jp_javaframe.cpp

```diff
@@ -0,0 +1,1116 @@
+/*****************************************************************************
+   Copyright 2004-2008 Steve Ménard
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+	   http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+
+ *****************************************************************************/
+#include <Python.h>
+#include "jpype.h"
+
+#define USE_JNI_VERSION JNI_VERSION_1_4
```

Comment:
What happens if this constant is changed in one unit (e.g. increased), while it remains the old value elsewhere?
Can we redirect the jni include via an extra header, which defines the used jni version once, and use it whenever jni is included?
If it does no harm to mix versions, I'd be fine with the current solution.

Reply:
There is one define here, and the other in jp_context

Reply:
This one is a stray and should be removed.  Only the one in jp_context is used.